base/pkg/libgit2/oid.jl

```diff
@@ -0,0 +1,116 @@
+const OID_RAWSZ = 20
+const OID_HEXSZ = OID_RAWSZ * 2
+const OID_MINPREFIXLEN = 4
+
+# immutable Oid
+#   id1::UInt8
+#   id2::UInt8
+#   ...
+#   id20::UInt8
+# end
+@eval begin
```

Comment:
The Oid generated code should now use an immutable type that wraps an ntuple of length 20